src/cosmicqc/scdataframe.py

```diff
@@ -0,0 +1,153 @@
+"""
+Defines a SCDataFrame class for use in coSMicQC.
+"""
+
+import pathlib
+from typing import Any, Dict, TypeVar, Union
+
+import pandas as pd
+
+# provide backwards compatibility for Self type in earlier Python versions.
+# see: https://peps.python.org/pep-0484/#annotating-instance-and-class-methods
+Self_SCDataFrame = TypeVar("Self_SCDataFrame", bound="SCDataFrame")
+
+
+class SCDataFrame:
+    """
+    A class to handle and load different types of data files into a pandas DataFrame.
+
+    This class can initialize with either a pandas DataFrame or a file path (CSV, TSV,
+    TXT, or Parquet). When initialized with a file path, it reads the data into a
+    pandas DataFrame.
+
+    Attributes:
+        data_source (str):
+            A string indicating the data source, either 'pd.DataFrame'
+            or the file path.
+        data (pd.DataFrame):
+            The loaded data in a pandas DataFrame.
+
+    Methods:
+        __call__():
+            Returns the underlying pandas DataFrame.
+        __repr__():
+            Returns representation of underlying pandas DataFrame.
+        __getattr__():
+            Returns underlying attributes of pandas DataFrame.
```

Comment:
```suggestion
            Returns the underlying attributes of the pandas DataFrame.
```

Or is the attributes of the SCDataFrame class? (I'll find out when I read further!)

Reply:
Thanks, that's accurate! I'll update.